okio/src/main/kotlin/okio/Buffer.kt

```diff
@@ -15,4 +15,131 @@
  */
 package okio
 
-expect class Buffer : BufferedSource, BufferedSink
+expect class Buffer : BufferedSource, BufferedSink {
+  override val buffer: Buffer
+
+  override fun emitCompleteSegments(): Buffer
+
+  override fun emit(): Buffer
+
+  override fun exhausted(): Boolean
+
+  override fun require(byteCount: Long)
+
+  override fun request(byteCount: Long): Boolean
+
+  override fun peek(): BufferedSource
+
+  override fun readByte(): Byte
+
+  override fun readShort(): Short
+
+  override fun readInt(): Int
+
+  override fun readLong(): Long
+
+  override fun readShortLe(): Short
+
+  override fun readIntLe(): Int
+
+  override fun readLongLe(): Long
+
+  override fun readDecimalLong(): Long
+
+  override fun readHexadecimalUnsignedLong(): Long
+
+  override fun readByteString(): ByteString
```

Comment:
None of these functions should need redeclare unless they're using covariant return types to change the return type to `Buffer`

Reply:
Oh, good call. I was getting build failures so pulled everything out without digging deep into specific methods. 

Covariant returns are a bit inconsistent due to the fact that `Buffer` omits return types in many places, so it's hard to say whether using covariant returns was intentional or not.

Reply:
I think it should just be for all the `write` methods to support chaining on the JVM without running into the checked exceptions from `BufferedSink`